### PR TITLE
SDK separation

### PR DESCRIPTION
**Pull request by tmbo**
**Proposed changes**:
- splits off custom logic, communicates with the custom actions over HTTP 

**Status (please check what you already did)**:
- [x] made PR ready for code review
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog

### CODE REVIEW

**Comment by akelad** [comment]
one thing: if you type '0' (export and quit) in interactive learning mode and you're running your custom action endpoint, shouldn't that also automatically kill it?

**Comment by akelad** [comment]
actually it also doesn't do that if you type `/stop`, not sure if that's something we should be handling or whether that's up to the user?

**Comment by akelad** [comment]
yeah true. Just a wee bit annoying, but i guess nothing we can do about it :P

**Comment by reza-ebrahimi** [comment]
When this big refactoring is merging?

File: tests/test_forms.py [hunk 1]
```diff
@@ -1,369 +0,0 @@
...
-from rasa_core.events import (
-    ActionExecuted,
-    BotUttered,
-    SlotSet,
-    UserUttered
-)
-from rasa_core.actions.forms import (
-    BooleanFormField,
-    EntityFormField,
-    FormAction,
```
Comment (akelad) [inline tests/test_forms.py]: FormActions are now part of the pySDK right, in which case we should remember to move these tests over there

File: CHANGELOG.rst [hunk 2]
```diff
@@ -11,10 +11,15 @@ This project adheres to `Semantic Versioning`_ starting with version 0.2.0.
 
 Added
 -----
+- script parameter ``--quiet`` to set the log level to ``WARNING``
```
[the remark this answers is not in the document]
Comment (tmbo) [reply CHANGELOG.rst]: yes good point, will go through the changes as well and take notes of them :+1:

File: docs/connectors.rst [hunk 3]
```diff
@@ -364,18 +395,25 @@ The command to run a ngrok instance for port 5002 for example would be:
 
 **Ngrok is only needed if you don't have a public IP and are testing locally**
 
-This will then give a output showing a https address that you need to supply for the interactive components request URL and for the incoming webhook and the address should be whatever ngrok supplies you with /webhook added to the end.  This basically takes the code running on your local machine and punches it through the internet at the ngrok address supplied.
+This will then give a output showing a https address that you need to
```
Comment (amn41) [inline docs/connectors.rst]: let's rewrite this to be simpler:

ngrok will great a https address for your computer, for example ``https://xxxxxx.ngrok.io`` .
For a facebook bot, your webhook address would them be ``https://xxxxxx.ngrok.io/webhooks/facebook/webhook``, for telegram ``https://xxxxxx.ngrok.io/webhooks/telegram/webhook``, etc.

File: docs/connectors.rst [hunk 4]
```diff
@@ -385,15 +423,14 @@ use the ``rasa_core.channels.channel.RestInput`` class as a template.
 The methods you need to implement are ``blueprint`` and ``name``. The method
 needs to create a Flask blueprint that can be attached to a flask server.
 
-This allows you to add REST endpoints to the server, the external
+This allows you to add `REST` endpoints to the server, the external
```
Comment (amn41) [inline docs/connectors.rst]: double backticks ?

**Comment by tmbo** [comment]
killer feature of this will be, that the stories in the docs are syntax highlighted 😆 

bildschirmfoto 2018-08-23 um 18 28 27

**Comment by tmbo** [comment]
@amn41 there is one page missing about running the action server - I'll add that on the train and merge afterwards. I'll ping you once I am done so you can give it a read.

**Comment by amn41** [comment]
sounds great! 🎉

**Comment by Ghostvv** [comment]
can we have this killer feature for pycharm?

**Comment by tmbo** [comment]
yes, but i am not to keen on doing the java implementation of that :D

**Comment by Ghostvv** [comment]
Let's ask community @JustinaPetr?

**Pull request merged**